builder/builder-next/controller.go

```diff
@@ -54,6 +42,19 @@ import (
 	bolt "go.etcd.io/bbolt"
 	"go.opentelemetry.io/otel/sdk/trace"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/builder/builder-next/adapters/containerimage"
+	"github.com/docker/docker/builder/builder-next/adapters/localinlinecache"
+	"github.com/docker/docker/builder/builder-next/adapters/snapshot"
+	"github.com/docker/docker/builder/builder-next/exporter/mobyexporter"
+	"github.com/docker/docker/builder/builder-next/imagerefchecker"
+	mobyworker "github.com/docker/docker/builder/builder-next/worker"
+	wlabel "github.com/docker/docker/builder/builder-next/worker/label"
+	"github.com/docker/docker/daemon/config"
+	"github.com/docker/docker/daemon/graphdriver"
+	units "github.com/docker/go-units"
+
 	"github.com/moby/buildkit/solver/pb"
 	"github.com/moby/buildkit/util/apicaps"
```

Comment:
Hm.. it looks like formatting and linting is not (yet) working as expected. ISTR having similar issues in the past when I tried a similar change; in this file, it's creating _more_ groups than the expected 3 (stdlib, external, local);


```go
import (
	"context"
// ....
	"time"

	ctd "github.com/containerd/containerd"
	"github.com/containerd/containerd/content/local"
// ....
	"go.opentelemetry.io/otel/sdk/trace"

	"github.com/docker/docker/api/types"
	"github.com/docker/docker/api/types/filters"
// ....
	"github.com/docker/docker/daemon/graphdriver"
	units "github.com/docker/go-units"

	"github.com/moby/buildkit/solver/pb"
	"github.com/moby/buildkit/util/apicaps"
)
```


I see it also grouped `github.com/docker/go-units` together with `github.com/docker/docker/....` imports (left a comment about that above).

I think the same applies to this changes as `gofumpt`; if we start to enforce this, ideally, we'd also have a `make` target (which could be together with the `gofumpt` one) that performs formatting with the correct options set, so that contributors can verify that all files are formatted correctly, and use that target to correct issues (if there are any), without having to modify their tooling / configuration.

Reply:
Hi @thaJeztah ,
I have tried gci and it seems to be working quite better. See [this PR ](https://github.com/moby/moby/pull/46880) .